### PR TITLE
Actually use free charge parameter in PerpendicularElectricFieldInterface

### LINKED ISSUE #22036: PerpendicularElectricFieldInterface erroneously does not use free charge
## Bug Description
The `PerpendicularElectricFieldInterface` object is supposed to apply a user-supplied free charge parameter when on an interface where free charge should be non-zero. 

## Steps to Reproduce
Try to use `PerpendicularElectricFieldInterface` with a `free_charge` parameter set to any number - it will have no impact on the solution. 

## Impact
This should be fixed so that expected operation of the interface kernel can be achieved.

### PR DESCRIPTION
**Pull request by cticenhour**
Closes #22036

### CODE REVIEW

**Comment by cticenhour** [comment]
@csdechant Apologies for the delay on this. Try out this branch to actually be able to use the free charge parameter in the PerpendicularElectricFieldInterface object. Working on getting you a new vector version of the PEC condition next.

File: modules/electromagnetics/test/tests/interfacekernels/electromagnetic_interfaces/tests [hunk 1]
```diff
@@ -23,12 +23,21 @@
     design = 'ParallelElectricFieldInterface.md PerpendicularElectricFieldInterface.md'
     issues = '#21075'
   []
+  [combined_props_zero_free_charge]
+    type = Exodiff
+    input = 'combined_props.i'
+    exodiff = 'combined_props_zero_free_charge_out.e'
+    cli_args = 'InterfaceKernels/perpendicular/free_charge=0 Outputs/file_base=combined_props_zero_free_charge_out'
+    requirement = 'The system shall calculate the appropriate interface condition dictated by Maxwells Equations for both perpendicular (subject to material properties) and parallel (equivalent) components of electromagnetic vector fields at an interface at the same time, with user-supplied material property parameters and zero free charge.'
```
Comment (grmnptr) [inline modules/electromagnetics/test/tests/interfacekernels/electromagnetic_interfaces/tests:31]:
```suggestion
    requirement = 'The system shall calculate the appropriate interface condition dictated by Maxwell's Equations for both perpendicular (subject to material properties) and parallel (equivalent) components of electromagnetic vector fields at an interface at the same time, with user-supplied material property parameters and zero free charge.'
```

**Review by grmnptr — COMMENTED** [review]
Looks good to me, just a few comments.

File: modules/electromagnetics/test/tests/interfacekernels/electromagnetic_interfaces/tests [hunk 2]
```diff
@@ -23,12 +23,21 @@
...
+    design = 'ParallelElectricFieldInterface.md PerpendicularElectricFieldInterface.md'
+    issues = '#21075'
```
[the remark this answers is not in the document]
Comment (cticenhour) [reply modules/electromagnetics/test/tests/interfacekernels/electromagnetic_interfaces/tests:33]: Maybe? This is effectively the same test that was here previously, even if the underlying input file was adjusted....But if  I'm making changes anyway to adjust a typo, might as well.

**Review by grmnptr — DISMISSED** [review]

**Review by grmnptr — APPROVED** [review]

**Pull request merged**